src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy

```diff
@@ -37,8 +39,15 @@ class IqPolicyEvaluatorUtil
                                                     final TaskListener listener)
   {
     try {
-      checkArgument(iqPolicyEvaluator.iqStage && iqPolicyEvaluator.iqApplication,
-          'Arguments iqApplication and iqStage are mandatory')
+      String applicationId = iqPolicyEvaluator.getIqApplication() != null ? iqPolicyEvaluator.
+          getIqApplication().applicationId : null
+
+      checkArgument(iqPolicyEvaluator.iqStage && applicationId, 'Arguments iqApplication and iqStage are mandatory')
+      checkArgument(IqUtil.verifyOrCreateApplication(NxiqConfiguration.serverUrl.toString(),
+          redentialsId: iqPolicyEvaluator.jobCredentialsId, context: run.parent, applicationId),
```

Comment:
sp. `redentials`... probably means something needs test coverage

Reply:
minor: I would have the call to IqUtil on a separate line to clarify the stack trace is anything goes wrong

Reply:
>sp. redentials... probably means something needs test coverage

>minor: I would have the call to IqUtil on a separate line to clarify the stack trace is anything goes wrong

Changed 1392026